### DISCUSSION TITLE
Second Class Developers

### DISCUSSION THREAD

**Discussion by r2d2Proton** [o3de/o3de#1922] [General]
Though Unreal Engine is open source, and does provide a lot, the public codebase is behind Epic Games internal codebase. Is that the case with the O3DE code?

There were numerous bugs in UE4 I could have easily fixed, some really annoying, but Epic took way too long to pull fixes. With that said, I am not holding multiple clones on GitHub and waiting for the pull request to be completed.

I know nothing about O3DE. So, I have to read the documentation, the contribute process, review the codebase and samples to form an opinion of the workflow. Hopefully it is streamlined.

Thank you for providing O3DE as open source.

**Comment by rgba16f** [comment]
Everyone is welcome to make changes.
Please fork the repo, make a branch and add your fixes in your fork, then create a PR from your branch to the o3de development branch.

**Reply by rgba16f** [reply]
PR acceptance and Merging privileges are given by the o3de organization.

**Comment by amznestebanpapp** [comment]
That is not currently the case for O3DE. Our fork is used only for branches before we merge them into `development` or into  `stabilization` branch. 
This is our fork: https://github.com/aws-lumberyard-dev/o3de

We dont fork `development` and `stabilization` branches. This means that e.g. for a bug, we branch o3de stabilization, fix the bug put the PR in o3de and merge it back to o3de/stabilization. We dont go through "an internal branch".
As a developer, we get a huge benefit on working at the public HEAD and push those bug fixes ASAP. 

For features, we need to be able to develop them in a branch. Bigger features/fixes will stay in branches until they are ready to become public. Likely we will experiment and find how others can see that WIP.

As a contributor, you should be able to put those fixes, moreover, we WELCOME that!!! Of course we all need to keep the code sane and with a certain level of quality so you will have to PR your change and get it approved by the relevant SIGs (as all our changes do) 
SIGs are not owned by a particular company, so your PR should find no objections as long as it meets the build requirements, it fixes the issue and has an acceptable level of quality (e.g. is not creating more bugs or deteriorating performance).

We encourage participation and contribution!

**Comment by r2d2Proton** [comment]
Thank you for the feedback. For the time being I am going to stay with UE4/UE5.

**Comment by ensignavenger** [comment]
One correction- Unreal Engine is not Open Source- the source code is available, but it is under a proprietary (closed) license.
